applications/NXmx.nxdl.xml

```diff
@@ -142,6 +142,46 @@
                         <dim index="4" value="k" required="false"/>
                     </dimensions>
                 </field>
+
+                <field name="data_scaling_factor" type="NX_FLOAT" optional="true">
+                    <doc>
+                        An optional scaling factor to apply to the values in ``data``.
+
+                        The elements in ``data`` are often stored as integers for efficiency reasons and need
+                        further correction, generating floats.  The two fields data_scaling_factor and
+                        data_offset allow linear corrections using the following convention:
+
+                        .. code-block::
+
+                                corrected_data = (data + offset) * scaling_factor
```

Comment:
[NON-BLOCKING]

Just as an observation, what (I believe) is being described here is a linear model for mapping between a continuous/observed values and the discrete/digitised signal.

A linear mapping is not the only approach.  Other approaches have been tried (in other fields) when it is desirable to focus more of the fidelity of digitised dynamic range in particular areas of interest; for real-world examples, see the [A-law](https://en.wikipedia.org/wiki/A-law_algorithm) and [μ-law](https://en.wikipedia.org/wiki/%CE%9C-law_algorithm) algorithms.

To be very clear: this comment is NON-BLOCKING.  It is *not* intended to delay or prevent the proposed change, which I think is reasonable, but rather to raise a question whether NeXus might need to support non-linear mappings in the future.

(My apologies if this is all "very obvious".  I'm not a domain expert, so I don't know to what extent is just part of "what everyone knows" :-) )